log/log.go

```diff
@@ -93,9 +131,12 @@ func WithPrefix(logger Logger, keyvals ...interface{}) Logger {
 //       returning a newly constructed context with a merged keyvals rather
 //       than simply wrapping the existing context.
 type context struct {
-	logger    Logger
-	keyvals   []interface{}
-	hasValuer bool
+	logger     Logger
+	keyvals    []interface{}
+	hasValuer  bool
```

Comment:
I think we should move `hasValuer` down a couple lines so that it is after `sKeyvals` and before `sHasValuer`. Putting all the `bool` fields consecutively at the end avoids padding bytes in the middle of the struct that the compiler will add to maintain the alignment requirements of the bigger slice fields.

Yes, it is a micro-optimization, and yes it is a bit less readable, but sometimes these things matter and it none of the fields are exported, so I am not too worried about the slight loss in readability in this case.

Reply:
I wasn't aware of this, Chris. Thanks for explaining, I found it fascinating! I've updated the PR.